community/kernel/src/main/java/org/neo4j/kernel/impl/factory/DataSourceModule.java

```diff
@@ -168,6 +170,11 @@ public DataSourceModule( final GraphDatabaseFacadeFactory.Dependencies dependenc
         DbmsOperations dbmsOperations = new NonTransactionalDbmsOperations( procedures );
         deps.satisfyDependency( dbmsOperations );
 
+        for ( SessionManager candidate : Service.load( SessionManager.class ) )
```

Comment:
Still, I cannot see any good alternatives. The service mechanism was used to avoid circular dependencies between kernel and bolt. I guess we would run into problem if another SessionManager is introduced as a Service, but that is not something that is done by accident, and anyway needs additional work on what that would actually mean. Happy to take suggestions over say slack, but otherwise I think we could take this solution for now, and solve later problems later.